### PR TITLE
Add collapsible sub-menu to sidebar

### PR DESCRIPTION
**Pull request by abrezuica**
Add collapsible sub-menu to sidebar

### CODE REVIEW

File: generators/app/templates/infrastructure/src/components/menu/MenuItem.js [hunk 1]
```diff
@@ -1,42 +1,56 @@
...
 import { useTranslation } from 'react-i18next';
+import { isNil } from "ramda";
 
 const useStyles = makeStyles(menuStyle);
 
-const MenuItem = ({ menu, drawerOpen, activeRoute }) => {
-  const classes = useStyles();
+const MenuItem = ({ menu, drawerOpen, activeRoute, isSubMenuItem, subMenuOpen, onToggleSubMenu }) => {
+  const { children, path, icon, text } = menu;
+  const isSubMenu = !isNil(children);
```
Comment (alexandra-c) [inline generators/app/templates/infrastructure/src/components/menu/MenuItem.js:15]: I think it's better to use falsy check here

File: generators/app/templates/infrastructure/src/components/menu/MenuItem.js [hunk 2]
```diff
@@ -1,42 +1,56 @@
...
-          <ListItemIcon className={classes.menuItemIcon}>
-            {menu.icon}
+        <Item {...itemProps} className={navLinkClasses}>
+          <ListItemIcon className={menuItemIconClasses}>
+            {icon}
           </ListItemIcon>
           <ListItemText
-            primary={text}
+            primary={translatedText}
+            secondary={isSubMenu && (subMenuOpen ? <ArrowDropUp className={classes.caret} /> : <ArrowDropDown className={classes.caret} />)}
```
Comment (alexandra-c) [inline generators/app/templates/infrastructure/src/components/menu/MenuItem.js:49]: what happens when this is false? did you check the console for any warnings?

File: generators/app/templates/infrastructure/src/components/menu/CollapsibleMenuItem.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+const CollapsibleMenuItem = ({ menu, drawerOpen, activeRoute }) => {
+  const [open, setOpen] = useState(false)
+  const toggleSubMenu = useCallback(() => setOpen(current => !current), [])
+
+  return (
+    <>
+      <MenuItem menu={menu} drawerOpen={drawerOpen} subMenuOpen={open} onToggleSubMenu={toggleSubMenu} />
+      <Collapse in={open} timeout="auto" unmountOnExit>
+        <List disablePadding>
+          {menu?.children?.map((subMenu, key) => (
```
Comment (alexandra-c) [inline generators/app/templates/infrastructure/src/components/menu/CollapsibleMenuItem.js:15]: I think you should add an example for a menu with aggregated children in menuConfig file

File: generators/app/templates/infrastructure/src/components/menu/MenuItem.js (hunk 2 above)
Comment (abrezuica) [reply generators/app/templates/infrastructure/src/components/menu/MenuItem.js:49]: it doesn't cause any warning, ListItemText receives undefined for secondary prop and doesn't draw anything in that place

Comment (alexandra-c) [reply generators/app/templates/infrastructure/src/components/menu/MenuItem.js:49]: are you sure? If **isSubMenu** is **false** I think the **secondary** prop will receive **false** not **undefined**. And **secondary** prop only accepts a **node** type value right?

Comment (abrezuica) [reply generators/app/templates/infrastructure/src/components/menu/MenuItem.js:48]: I was wrong, it's false :d but it seems boolean is considered to be a react node type that is ignored and this is why no warning is thrown and nothing is drawn for secondary in the UI
https://reactnative.dev/docs/react-node

**Review by alexandra-c — CHANGES_REQUESTED** [review]

**Review by alexandra-c — APPROVED** [review]

**Pull request merged**